src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/visualeditor/VisualEditorPageObject.java

```diff
@@ -476,29 +476,34 @@ public VisualEditorEditTemplateDialog openEditTemplateDialog() {
   }
 
   public VisualEditorPageObject clickInsertToolButton() {
+    wait.forElementClickable(insertDropdownMenuButton);
     insertDropdownMenuButton.click();
     return this;
   }
 
   public VisualEditorPageObject clickInsertInfoboxFromInsertToolMenu() {
-    wait.forElementVisible(infoboxInDropdownMenu);
+    wait.forElementClickable(infoboxInDropdownMenu);
     infoboxInDropdownMenu.click();
     return this;
   }
 
   public VisualEditorPageObject selectInfoboxTemplate(int i) {
-    infoboxTemplatesList.get(i).click();
+    WebElement element = infoboxTemplatesList.get(i);
+    wait.forElementClickable(element);
+    element.click();
     return this;
   }
 
   public VisualEditorPageObject typeInParameterField(int i, String parameter) {
-    parametersFieldList.get(i).click();
-    parametersFieldList.get(i).sendKeys(parameter);
+    WebElement element = parametersFieldList.get(i);
+    wait.forElementClickable(element);
+    element.click();
+    element.sendKeys(parameter);
```

Comment:
can we add here more descriptive name instead of `element`? Maybe `parameterInput` or `input` or `field`?